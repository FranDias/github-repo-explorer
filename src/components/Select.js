import React, { Component } from "react";

class App extends Component {

    render (
        <select onChange={this.selectSortCount} name="selectSortOption">
        {sortOptions.map(choice => (
          <option key={choice} value={choice}>
            Sort by {choice}
          </option>
        ))}
      </select>
    )
}

export default Select;