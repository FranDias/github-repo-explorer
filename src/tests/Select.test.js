import React from "react";
import Select from "../components/Select";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const testFunction = () => {};

const selectOptions = {
  name: "testSelect",
  onChange: testFunction,
  options: ["zero", "one", "two"]
};

describe("Dropdown Select", () => {
  it("renders a select element", () => {
    const component = renderer.create(<Select {...{ selectOptions }} />);
    expect(component).toMatchSnapshot();
  });

  describe("props", () => {
    Object.keys(selectOptions).map(key => {
      it(`renders ${key}`, () => {
        const component = mount(
          <Select {...{ [key]: selectOptions[key] }} />
        );
        expect(component.find(Select).props()[key]).toEqual(selectOptions[key]);
      });
    });
  });
});
