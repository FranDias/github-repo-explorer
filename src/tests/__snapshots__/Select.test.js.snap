// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Select renders a select element 1`] = `
<select
  onChange={[Function]}
>
  <option
    value="no options provied"
  >
    Sort by 
    no options provied
  </option>
</select>
`;
